doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,158 @@
+  used by the receiving side which is running in another thread.
+- To unsubscribe, the NDRegisterAccessor is calling AsyncAccessorManager::unsibscribe() with its TransferElementID from
+  the destructor.
+  For the unsubscribe mechanism to work, the AsyncAccessorManager can only hold a weak pointer.
+  If it would hold a shared pointer, the destructor would never be called because this instance survives,
+  even if the user code has deleted all copies of the AsyncNDRegisterAccessor.
+- To manage the weak pointer and to have an entry object for the implementation, the helper class
+  AsyncVariable has been introduced.\ref asyncAccessor_comment_AsyncVariable "(*)"
+  It contains the weak pointer as a private variable and provides member functions to access the object behind
+  the weak pointer.
+  This ensures two points which otherwise would be left to the implementation, which would be error prone:
+  - The weak pointer is locked before accessing the object.
+  - There is a race condition that the locked object could have a nullptr if the shared pointer is currently being
+     destructed while the lock is being acquired. This is checked as well.
+- The void notification queue in the TransferElement base class is set up as a continuation of the data transport queue,
+  such that the read operations of the notification queue trigger the filling of the user buffer.
+
+\anchor asyncAccessor_comment_AsyncVariable (*) There is AsyncVariable as an untyped base class and AsyncVariableImpl,
+  which is templated to a user type. Like this pointers to variables with different user types can be stored in one list.
+
+\subsection AsyncNDRegisterAccessor_usage Interface for implementing backends
+
+While the AsyncNDRegisterAccessor itself is complete and universal, the AsyncAccessorManager and
+AsyncVariable have pure virtual functions which must be provided by the individual
+backend implementations:
+- AsyncVariable::getNumberOfChannels(), AsyncVariable::getNumberOfSamples(), AsyncVariable::getUnit(),
+  AsyncVariable::getDescription() and AsyncVariable::isWriteable() must be provided by deriving from AsyncVariable or
+  AsyncVariableImpl. The information must directly be provided by the backend.
+  The generating code cannot rely on the catalogue here (which might not even be filled yet when the accessor
+  is being created).\ref asyncAccessor_comment_AsyncVariable "(*2)"
+- The 'virtual function template' AsyncAccessorManager::createAsyncVariable() must return an AsyncVariable
+  implementation with the implemented accessor-describing functions from the previous bullet point, such that the
+  accessor manager can create the AsyncNDRegisterAccessor.
+  When createAsyncVariable() returns the AsyncVariable object, the weak pointer to the AsyncNDRegisterAccessor is not
+  set yet. The reason is that the backend specific code should not have the subscription code, which is the same for
```

Comment:
I think the reasoning here is more distracting than helping. Once the reader has understood enough context to be able to comprehend what you are saying here, the reason already is self-evident.

Reply:
I tried to shorten it. Section still is not very catchy